common/lib/client/auth.js

```diff
@@ -534,6 +519,144 @@ var Auth = (function() {
 		}
 	};
 
+	Auth.prototype._saveBasicOptions = function(defaultTokenParams, authOptions) {
+		this.method = 'basic';
+		this.key = authOptions.key;
+		this.basicKey = toBase64(authOptions.key);
+		this.tokenParams = defaultTokenParams || {};
+		this.authOptions = authOptions || {};
+		delete this.authOptions.force;
+		if('clientId' in authOptions) {
+			this._userSetClientId(authOptions.clientId);
+		}
+	}
+
+	Auth.prototype._saveTokenOptions = function(tokenParams, authOptions) {
+		this.method = 'token';
+
+		this.tokenParams = tokenParams || this.tokenParams || {};
+
+		/* If an authOptions object is passed in that contains new auth info (ie
+		* isn't just {force: true} or something), it becomes the new default, with
+		* the exception of the force attribute (RSA10g), which is set anew on each
+		* call to authorise (defaulting to false) */
+		this.force = false;
+		if(authOptions) {
+			this.force = authOptions.force;
+
+			if(containsAuthInfo(authOptions)) {
+				this.authOptions = authOptions || {};
```

Comment:
Of course this is why it was a bad idea all along to have `authorise()` persist the overrides, and a `setOptions()`, which fully supersedes the previously set options, would have been more intuitive IMO :)

Merging (and then persisting) is inconsistent with the other APIs and doesn't let you unset things once they're set in java and similar languages. Unsetting them by including `authURL: undefined` etc is a bit horrible anyway.

So, if we're not going to have a `setOptions()` which explicitly and fully supersedes the set options, then what @SimonWoolf has done here (which is piecemeal and arbitrary) may well deal with the typical use-cases as well as possible.

> believe we have merged in all other implementations

It's not as if it's a huge change, so unfortunately I think that's not a good enough reason by itself to go further along the road of making an inferior API.